internal/elasticsearch/v8/search.go

```diff
@@ -0,0 +1,62 @@
+// Copyright © 2022 Meroxa, Inc. and Miquido
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v8
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/conduitio-labs/conduit-connector-elasticsearch/internal/elasticsearch/api"
+
+	"github.com/elastic/go-elasticsearch/esapi"
+)
+
+// Search calls the elasticsearch search api and retuns SearchResponse read from an index.
+func (c *Client) Search(ctx context.Context, index string, offset, size *int) (*api.SearchResponse, error) {
+	// Create the search request
+	req := esapi.SearchRequest{
+		Index: []string{index},
+		Body: strings.NewReader(`{
+			"query": {
+				"match_all": {}
+			}
+		}`),
+		From: offset,
+		Size: size,
+	}
```

Comment:
I believe we'll something to sort the results so that, when we restart the source connector, we can start from exactly where we left: https://www.elastic.co/guide/en/elasticsearch/reference/current/paginate-search-results.html#search-after

Reply:
Elasticsearch internally stores an integer _seq_no and returns a sorted list based on this integer. When we restart the connector, the position map helps us read values from that point onward, ensuring we only retrieve new records.

Reply:
Updated, used search_after, Thanks!